dj/api/nodes.py

```diff
@@ -213,6 +213,16 @@ def read_node(name: str, *, session: Session = Depends(get_session)) -> NodeOutp
     return node  # type: ignore
 
 
+@router.delete("/nodes/{name}/", status_code=204)
+def delete_node(name: str, *, session: Session = Depends(get_session)):
+    """
+    Delete the specified node.
+    """
+    node = get_node_by_name(session, name, with_current=True)
+    session.delete(node)
```

Comment:
In this function we need to get all of the downstream nodes and switch their status to invalid. That will probably be a common action so maybe a helper function that just invalidates all downstream nodes will probably be useful.

I'm not sure how to handle if this is a dimension node and there are columns on other nodes linking to it. On the one hand we could simply delete the link as well but that makes deleting a dimension node potentially have a very large blast radius.

This might need some more discussion so feel free to leave these for a follow-up PR.

Reply:
Good point, I forgot about invalidation! Let me add that.

Although regarding dimension nodes having a large blast radius - I'm not sure that's the case. Wouldn't we just remove the links to that dimension for any nodes that have a field linked to that dim node? The downstream nodes (via dimension links) themselves don't need to be invalidated. However, if it's a case of downstream nodes that reference that dimension node as a child, then they'll need to be invalidated, but I don't think there would be that many of those in the system.

Reply:
Yeah that's true, I was thinking of a scenario where someone naively thinks, "oh let me just delete this dimension and recreate it from scratch", not realizing that all of the potential links will be dropped and need to be re-added. Maybe that's not as much as a concern as it seems. 🤔

Reply:
Wow, this a great question... my gut feeling for dimension nodes deletion is that we should not invalidate any of the dim/connected nodes but we may simply consider a missing dimension link as a run time error that we could handle gracefully (or not) depending on the situation. Either way I would treat this linkage loosely so as to not cause a node panic/invalidation for this reason. The other side (parent-child nodes) is clear to follow the valid pattern but perhaps dimension links can be a softer touch, which would allow someone to just drop a dimension node and recreate it. Just a thought. I agree we should run this by others, I can see an argument for keeping it as strict as the parent-child node relationship.